src/routes/(inner)/elements/chat/+page.svelte

```diff
@@ -0,0 +1,563 @@
+				<svelte:fragment slot="preview">
+					<div class="input-group input-group-divider grid-cols-[auto_1fr_auto] rounded-container-token">
+						<button class="input-group-shim">+</button>
+						<textarea
+							bind:value={currentMessage}
+							class="bg-transparent border-0 ring-0"
+							name="prompt"
+							id="prompt"
+							placeholder="Write a message..."
+							rows="1"
+						/>
+						<button class="variant-filled-primary">Send</button>
+					</div>
+				</svelte:fragment>
+				<svelte:fragment slot="source">
+					<CodeBlock language="ts" code={`let currentMessage = '';`} />
+					<CodeBlock
+						language="html"
+						code={`
+<div class="input-group input-group-divider grid-cols-[auto_1fr_auto] rounded-container-token">
+	<button class="input-group-shim">+</button>
+	<textarea
+		bind:value={currentMessage}
+		class="bg-transparent border-0 ring-0"
+		name="prompt"
+		id="prompt"
+		placeholder="Write a message..."
+		rows="1"
+	/>
+	<button class="variant-filled-primary">Send</button>
+</div>
+					`}
+					/>
+				</svelte:fragment>
+			</DocsPreview>
+		</section>
+		<section class="space-y-4">
+			<h2>Scroll to Bottom</h2>
+			<p>Use <code class="code">bind:this</code> to bind your scrollable chat element.</p>
+			<CodeBlock language="ts" code={`let elemChat: HTMLElement;`} />
+			<CodeBlock language="html" code={`<div bind:this={elemChat} class="overflow-y-auto">(chat)</div>`} />
+			<p>
+				use Javascript's <a
+					class="anchor"
+					href="https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTo"
+					target="_blank"
+					rel="noreferrer">scrollTo</a
+				>
+				method to scroll the binded element to the bottom. Enable smooth scroll with
+				<code class="code">behavior: 'smooth'</code>.
+			</p>
+			<CodeBlock
+				language="ts"
+				code={`
+function scrollChatBottom(behavior?: ScrollBehavior): void {
+	elemChat.scrollTo({ top: elemChat.scrollHeight, behavior });
+}
+				`}
+			/>
+		</section>
+		<section class="space-y-4">
+			<h2>Add a Message</h2>
+			<p>Ensure you have a set of messsage data. We'll provide an example below.</p>
+			<CodeBlock
+				language="ts"
+				code={`
+let messages = [
+	{
+		id: 0,
+		host: true,
+		avatar: 48,
+		name: 'Jane',
+		timestamp: 'Yesterday @ 2:30pm',
+		message: lorem,
+		color: 'variant-soft-primary'
+	},
+	// ...
+];
+					`}
+			/>
+			<p>Bind to your prompt textarea to be able to capture the contents on demand.</p>
+			<CodeBlock language="ts" code={`let currentMessage = '';`} />
+			<CodeBlock
+				language="html"
+				code={`
+<textarea
+	bind:value={currentMessage} name="prompt" id="prompt"
+	placeholder="Write a message..."
+/>
+					`}
+			/>
+			<p>Use the following function to append a new message to the <code class="code">messages</code> array.</p>
+			<CodeBlock
+				language="ts"
+				code={`
+function addMessage(): void {
+	const newMessage = {
+		id: messages.length,
+		host: true,
+		avatar: 48,
+		name: 'Jane',
+		timestamp: new date(),
```

Comment:
We're changing from using natural English to the `Date` here. it might be worth mentioning that the natural language "time since" can be achieved with [DayJS](https://day.js.org/docs/en/display/from). Might also want to change it to use a serializable time format instead of the Date object.

Reply:
As discussed on Discord last night, I think introducing third party tools would be out of scope here. In most cases getting the current timestamp is simple enough via a formatted JS data object. We provide `getCurrentTimestamp()` in the source if folks really need a turnkey solution for this.

Overall this is well outside of scope for this feature though. The focus is "how do I use Skeleton and Tailwind to create a chat UI" and less how do I build a fully functional chat UI. The answer to that will differ greatly on implementation.